### PR TITLE
Add option to re-enable libstdc++ at avr

### PR DESCRIPTION
**Pull request by QBos07**
Signed-off-by: Quentin Boswank <qubos@outlook.de>

### CODE REVIEW

**Comment by cpackham** [comment]
We'd be better of doing #808 then C++ would be enabled bu default.

**Comment by QBos07** [comment]
It's not the C++ support. It's the library that's intended to be disabled with avr and to use a freestanding C++ with the really limited space on this devices.

**Comment by QBos07** [comment]
any updates here @cpackham ?

File: config/cc/gcc.in [hunk 1]
```diff
@@ -54,6 +54,15 @@ config CC_GCC_ENABLE_CXX_FLAGS
...
       after the '=' sign.
 
+config CC_GCC_EXTRA_LIBSTDCXX
+    bool "Re-enable libstdcxx"
+    default n
+    depends on LIBC_AVR_LIBC
+    select CC_CORE_PASS_2_NEEDED
+    help
+      libstdcxx is normaly disabled on avr systems, so yout can define your on datatypes
+      like <type-traits>. This re-enable this.
```
Comment (cpackham) [inline config/cc/gcc.in:64]: Couple of bit of a typos here "normaly" and "yout". I think perhaps we can write this as

```
      libstdcxx is normally disabled on avr systems due to size limitations. Set this option to re-enable
      libstdcxx support.
```

**Review by cpackham — COMMENTED** [review]
In terms of a commit message I think we need a better justification for this change. I had to do a bit of searching to figure out what the situation is with C++ support on AVR. I think I understand now but a good commit message will help future readers.

I'd also prefer sign-off from a real name.

**Comment by QBos07** [comment]
Thanks for the response. I will give it another pass in 3 days when I’m home.
Am 15.07.2023 um 06:58 schrieb Chris Packham ***@***.******@***.***>>:


@cpackham commented on this pull request.

In terms of a commit message I think we need a better justification for this change. I had to do a bit of searching to figure out what the situation is with C++ support on AVR. I think I understand now but a good commit message will help future readers.

I'd also prefer sign-off from a real name.

________________________________

In config/cc/gcc.in<https://github.com/crosstool-ng/crosstool-ng/pull/1569#discussion_r1264326445>:

> @@ -54,6 +54,15 @@ config CC_GCC_ENABLE_CXX_FLAGS
       Note: just pass in the option _value_, that is only the part that goes
       after the '=' sign.

+config CC_GCC_EXTRA_LIBSTDCXX
+    bool "Re-enable libstdcxx"
+    default n
+    depends on LIBC_AVR_LIBC
+    select CC_CORE_PASS_2_NEEDED
+    help
+      libstdcxx is normaly disabled on avr systems, so yout can define your on datatypes
+      like <type-traits>. This re-enable this.


Couple of bit of a typos here "normaly" and "yout". I think perhaps we can write this as

      libstdcxx is normally disabled on avr systems due to size limitations. Set this option to re-enable
      libstdcxx support.

**Comment by QBos07** [comment]
I need a bit more time. Windows blown up my Linux boot.

**Comment by QBos07** [comment]
Fixed the spelling.
Gave proper commit message and real sign off.

**Pull request merged**